tests/unit/phpunit/include/SugarFolders/SugarFolderTest.php

```diff
@@ -0,0 +1,725 @@
+
+        $parentFolderOne->copyBean($parentFolderOne->id, $parentFolderTwo->id, $bean->id, $bean->module_name);
+
+        $existInFolderOne = $parentFolderOne->checkEmailExistForFolder($bean->id);
+        $existInFolderTwo = $parentFolderTwo->checkEmailExistForFolder($bean->id);
+
+        $this->assertTrue($existInFolderOne);
+        $this->assertTrue($existInFolderTwo);
+
+        $parentFolderOne->deleteEmailFromAllFolder($bean->id);
+
+        $existInFolderOne = $parentFolderOne->checkEmailExistForFolder($bean->id);
+        $existInFolderTwo = $parentFolderTwo->checkEmailExistForFolder($bean->id);
+
+        $this->assertFalse($existInFolderOne);
+        $this->assertFalse($existInFolderTwo);
+
+        $this->popState();
+    }
+
+    public function testGetUserFolders()
+    {
+        $this->pushState();
+
+        $user = new User();
+        $user->id = 1;
+
+        $parentFolderOne = new SugarFolder($user);
+
+        $fields = array(
+            'name'             => 'Parent Folder',
+            'parent_folder'    => ''
+        );
+
+        $saved = $parentFolderOne->setFolder($fields);
+
+        $this->assertTrue($saved);
+
+        $arrayOfEmailBeans = array();
+
+        for ($i = 0; $i < 10; $i++) {
+            $arrayOfEmailBeans[$i] = BeanFactory::newBean('Emails');
+            $arrayOfEmailBeans[$i]->save();
+
+            $parentFolderOne->addBean($arrayOfEmailBeans[$i]);
+        }
+
+        $childFolder = new SugarFolder($user);
+
+        $fields = array(
+            'name'             => 'Child Folder',
+            'parent_folder'    => $parentFolderOne->id
+        );
+
+        $subChildFolderOne = new SugarFolder($user);
+
+        $fields = array(
+            'name'             => 'Sub Child Folder One',
+            'parent_folder'    => $childFolder->id
+        );
+
+        $subChildFolderTwo = new SugarFolder($user);
+
+        $fields = array(
+            'name'             => 'Sub Child Folder Two',
+            'parent_folder'    => $childFolder->id
+        );
+
+        $anotherChildFolder = new SugarFolder($user);
+
+        $fields = array(
+            'name'             => 'Another Child Folder',
+            'parent_folder'    => $parentFolderOne->id
+        );
+
+        $subs = array($anotherChildFolder->id, $parentFolderOne->id, $childFolder->id, $subChildFolderOne->id, $subChildFolderTwo->id);
+        $parentFolderOne->setSubscriptions($subs);
+
+        $email = new Email();
+        $email->email2init();
+
+        $ie = new InboundEmail();
+        $ie->email = $email;
+
+        $rootNode = new ExtNode('', '');
+
+        $folderOpenState = $user->getPreference('folderOpenState', 'Emails');
+        $folderOpenState = empty($folderOpenState) ? '' : $folderOpenState;
+
+        try {
+            $parentFolderOne->getUserFolders($rootNode, sugar_unserialize($folderOpenState), null, true);
+        } catch(SugarFolderEmptyException $e) {
+
```

Comment:
:scream: Ok, this test passing when your function throws a SugarFolderEmptyException. but also passing if it doesn't?!